src/Tools/dotnet-gcdump/CommandLine/ReportCommandHandler.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.CommandLine;
+using System.CommandLine.Binding;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Tools.Common;
+
+namespace Microsoft.Diagnostics.Tools.GCDump
+{
+    internal static class ReportCommandHandler
+    {
+        delegate Task<int> ReportDelegate(CancellationToken ct, IConsole console, FileInfo file = null, int? processId = null, ReportType reportType = ReportType.HeapStat);
+        
+        public static Command ReportCommand() =>
+            new Command(
+                name: "report",
+                description: "Generate report into stdout from a previously generated gcdump or from a running process.")
+            {
+                // Handler
+                HandlerDescriptor.FromDelegate((ReportDelegate) Report).GetCommandHandler(),
+                // Options
+                FileNameOption(), ProcessIdOption(), ReportTypeOption()
+            };
+
+        private static Task<int> Report(CancellationToken ct, IConsole console, FileInfo file = null, int? processId = null, ReportType type = ReportType.HeapStat)
+        {
+            //
+            // Validation
+            //
+            if (file == null && !processId.HasValue)
```

Comment:
Just a reminder to make the `-f|--file` option the positional argument for this command.